libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -208,6 +211,9 @@ constructor(
         fasterRouteController = FasterRouteController(directionsSession, tripSession, logger)
         routeRefreshController = RouteRefreshController(directionsSession, tripSession, logger)
         routeRefreshController.start()
+
+        arrivalProgressObserver = ArrivalProgressObserver(tripSession)
+        tripSession.registerRouteProgressObserver(arrivalProgressObserver)
```

Comment:
https://github.com/mapbox/mapbox-navigation-android/blob/5c1175f118f44aa6543c218403bdc69ef26dbeb7/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L302

I didn't unregister this one because all of them are unregistered onDestroy. Which is interesting, how does the foreground activity continue to work?

Reply:
Ohh, that's right 👍 Sorry for the confusion!